server/etcdserver/version/version.go

```diff
@@ -65,3 +77,80 @@ func detectStorageVersion(lg *zap.Logger, tx backend.ReadTx) (*semver.Version, e
 	copied := V3_5
 	return &copied, nil
 }
+
+func migrateOnce(lg *zap.Logger, tx backend.BatchTx, current semver.Version, upgrade bool) (*semver.Version, error) {
+	var target semver.Version
+	if upgrade {
+		target = semver.Version{Major: current.Major, Minor: current.Minor + 1}
+	} else {
+		target = semver.Version{Major: current.Major, Minor: current.Minor - 1}
+	}
+	ms := migrations(lg, current, target)
```

Comment:
Nit: maybe instead lets name this newMigration, wdyt?
```suggestion
	ms := newMigration(lg, current, target)
```

Reply:
This functions returns list of boltdb modifications "migrations" that need to be executed. Changing function name to `newMigration` implies that single object is returned. We could add a new structure, but then we would need to change the name of interface.

Reply:
maybe `getMigrations`? Just a thought. More of a nit really.

Reply:
Renamed `migration` to `migtionAction` and introduced a `newMigration` function like you proposed. I think it's more clearer as we don't reuse same name. 

`migration` struct is a change between two neighboring versions (v3.5 and v3.6). `Migration` is a collection of actions that implement `migrationAction` interface. Action is a unit of change like add a new field.